insertion_sort.go

```diff
@@ -11,7 +9,11 @@ type LessFunc func(a, b int) bool
 
 // InsertionSort rearranges the list to become sorted.
 func InsertionSort(l *list.List, less LessFunc) {
-	// TODO
+	for n := l.Front().Next(); n != nil; n = n.Next() {
+		for k := n; k != l.Front() && k.Previous().Value > k.Value; k = k.Previous() {
+			l.SwapElements(k.Previous().Value, k.Value)
+		}
+	}
 }
 
 // SortedInsert inserts a new element with a given value in a sorted list
```

Comment:
Where is SortedInsert implementation?

Reply:
I don't need it in my implementation, it works without this additional function (except for some cases with repeating numbers, but I planned to fix it later). But since I can't use 'previous' and my SwapElements function, I will probably need to implement it.